src/XRP/xrp-issued-currency.ts

```diff
@@ -0,0 +1,36 @@
+import bigInt, { BigInteger } from 'big-integer'
+import XRPCurrency from './xrp-currency'
+import { IssuedCurrencyAmount } from '../generated/web/org/xrpl/rpc/v1/amount_pb'
+
+/*
+ * An issued currency on the XRP Ledger
+ * @see: https://xrpl.org/basic-data-types.html#specifying-currency-amounts
+ */
+export default class XRPIssuedCurrency {
+  public static from(
+    issuedCurrency: IssuedCurrencyAmount,
+  ): XRPIssuedCurrency | undefined {
+    const currency = issuedCurrency.getCurrency()
+    let xrpCurrency
+    if (currency) {
+      xrpCurrency = XRPCurrency.from(currency)
+    }
+    let value
```

Comment:
nit: I'd probably use some vertical spacing to break up this function after you set each variable.